core/common/src/main/java/alluxio/security/authentication/SaslStreamServerDriver.java

```diff
@@ -111,6 +120,13 @@ public void onError(Throwable throwable) {
       LOG.warn("Closing authenticated channel: {} due to error: {}", mChannelId, throwable);
       mAuthenticationServer.unregisterChannel(mChannelId);
     }
+    if (mSaslServerHandler != null) {
+      try {
+        mSaslServerHandler.close();
+      } catch (IOException exc) {
```

Comment:
Do we also need to close the handler when `SaslStreamServerDriver.close()` is called?

Reply:
No, SaslServer will be closed before reaching there for both success and failure paths.

Reply:
I couldn't find any cleanar way for serverdriver to close its handler. Now closing it during close call as well.